client/src/details/UrlBuilder.h

```diff
@@ -42,46 +42,66 @@ class UrlBuilder
     /**
      * @brief Set the scheme for the URL
      * @param scheme The scheme to set for the URL Ex: Https
+     * @return The reference to the current object
      */
-    void SetScheme(Scheme scheme);
+    UrlBuilder& SetScheme(Scheme scheme);
 
     /**
      * @brief Set the host for the URL
      * @param host The host to set for the URL. Ex: www.example.com
      * @throws SFSException if the string is invalid
+     * @return The reference to the current object
      */
-    void SetHost(const std::string& host);
+    UrlBuilder& SetHost(const std::string& host);
 
     /**
      * @brief Set a path to the URL
      * @param path The path to set for the URL. Ex: index.html
      * @param encode If true, the path will be URL encoded
      * @throws SFSException if the string is invalid
+     * @return The reference to the current object
      */
-    void SetPath(const std::string& path, bool encode = false);
+    UrlBuilder& SetPath(const std::string& path, bool encode = false);
+
+    /**
+     * @brief Append a path to the URL
+     * @param path The path to be appended for the URL. Ex: index.html
+     * @param encode If true, the new path element will be URL encoded, including forward slashes
+     * @throws SFSException if the string is invalid
+     * @return The reference to the current object
+     */
+    UrlBuilder& AppendPath(const std::string& path, bool encode = false);

```

Comment:
as I mentioned in another review, here (and SetPath), avoid bool parameters (and especially default args!) and provide explicit methods, e.g.

* SetPath
* SetPathEncoded
* AppendPath
* AppendPathEncoded

these can each be inline calls into shared methods, but i've seen too many issues with relying on a bool and someone getting it wrong (`AppendPath(path)` vs `AppendPath(path, true)`), not to mention that anytime a parameter comment needs a comment to clarify (e.g. `AppendPath(path, true /*encode*/)`), it's a code smell.